platform/android/java/src/org/godotengine/godot/GodotView.java

```diff
@@ -363,10 +363,43 @@ public void run() {
 				});
 			}
 			return true;
-		};
+		} else if ((event.getSource() & InputDevice.SOURCE_STYLUS) == InputDevice.SOURCE_STYLUS && event.getAction() == MotionEvent.ACTION_HOVER_ENTER) {
+
+			final int x = Math.round(event.getX());
+			final int y = Math.round(event.getY());
+			queueEvent(new Runnable() {
+				@Override
+				public void run() {
+					GodotLib.hover(MotionEvent.ACTION_HOVER_ENTER, x, y);
+				}
+			});
+			return true;
+		} else if ((event.getSource() & InputDevice.SOURCE_STYLUS) == InputDevice.SOURCE_STYLUS && event.getAction() == MotionEvent.ACTION_HOVER_EXIT) {
+
+			final int x = Math.round(event.getX());
+			final int y = Math.round(event.getY());
+			queueEvent(new Runnable() {
+				@Override
+				public void run() {
+					GodotLib.hover(MotionEvent.ACTION_HOVER_EXIT, x, y);
+				}
+			});
+			return true;
+		} else if ((event.getSource() & InputDevice.SOURCE_STYLUS) == InputDevice.SOURCE_STYLUS && event.getAction() == MotionEvent.ACTION_HOVER_MOVE) {
+
+			final int x = Math.round(event.getX());
+			final int y = Math.round(event.getY());
+			queueEvent(new Runnable() {
+				@Override
+				public void run() {
+					GodotLib.hover(MotionEvent.ACTION_HOVER_MOVE, x, y);
+				}
+			});
+			return true;
+		}
```

Comment:
There's a lot of repetition in this section. You can simplify by having the following:
```
if ((event.getSource() & InputDevice.SOURCE_STYLUS) == InputDevice.SOURCE_STYLUS) {
                        final int x = Math.round(event.getX());
			final int y = Math.round(event.getY());
			queueEvent(new Runnable() {
				@Override
				public void run() {
					GodotLib.hover(event.getAction(), x, y);
				}
			});
			return true;
}
```